ledger/complete/wal/checkpoint_v6_reader.go

```diff
@@ -365,10 +371,12 @@ func getNodeByIndex(subtrieNodes [][]*node.Node, topLevelNodes []*node.Node, ind
 		return nil, fmt.Errorf("top trie [0] %v isn't nil", topLevelNodes[0])
 	}
 
-	if offset >= uint64(len(topLevelNodes)-1) {
-		return nil, fmt.Errorf("can not find node by index: %v in subtrieNodes %v, topLevelNodes %v", index, subtrieNodes, topLevelNodes)
+	nodePos := index - totalSubTrieNodeCount
+
+	if nodePos >= uint64(len(topLevelNodes)) {
```

Comment:
Yeah, that makes sense. I can remove the sanity check above.

But I think I should still keep the check here, because it's essentially to replace [the existing check here](https://github.com/onflow/flow-go/blob/master/ledger/complete/wal/checkpointer.go#L906-L908)